tests/checked_set_test.py

```diff
@@ -71,6 +72,7 @@ def test_create():
 def test_evolver_returns_same_instance_when_no_updates():
     x = Naturals([1, 2])
     assert x.evolver().persistent() is x
+    assert x.evolver().add(1).update([1, 2]).persistent() is x
```

Comment:
What about if items are added and then deleted before the evolver is closed?
Worth adding a test that that also returns the exact same instance?